### DISCUSSION TITLE
browser back - history.push remount page.

### DISCUSSION THREAD

**Discussion by onopko81** [vercel/next.js#17658] [Help]
Hi, I would need to change url with history.push, but doing a back from the browser I would have to remount the page. It's possible? I should stay on the same page and simulate the page change, but when I back up from the browser it doesn't re-mount the page. how could I do?

thanks a lot.

**Comment by jamesmosier** [comment]
Hi @onopko81. Could you explain a bit more what you are meaning by "mount" and "re-mount" of the page?

Next has built in methods for pushing history onto the history stack: https://nextjs.org/docs/api-reference/next/router#routerpush

```js
import { useRouter } from 'next/router'
const router = useRouter()
router.push('/your-page')
```

**Reply by jamesmosier** [reply]
> Hi,
is it possible to make a history.push ('link') change the link in the browser but not load the new page? my problem is that then doing back from browser I would like to render the starting link again. I don't know if I was clear

That wouldn't be native Next.js functionality. Anytime a route changes Next loads that new page.

If you wanted to do this with native JavaScript methods you could do `window.history.pushState(state, title, url)` ([docs](https://developer.mozilla.org/en-US/docs/Web/API/History/pushState)) but I'm not sure how well that'd work within Next.

**Comment by onopko81** [comment]
Hi,
is it possible to make a history.push ('link') change the link in the browser but not load the new page? my problem is that then doing back from browser I would like to render the starting link again. I don't know if I was clear

**Comment by onopko81** [comment]
I needed this theoretically to make a transition animation between two pages, while remaining within the first page. i tried javascript native pushstate but nothing, it doesn't work.

**Comment by TruRooms** [comment]
I am trying to tackle a similar issue.

We have a search results page where a user can filter results which are loaded via fetch. 

Ideally, the query string needs to be updated as the user changes their filters. That way they can share a link, or once they visit a result, hit the back button and not have to redo their filters.

The caveat being, that the page needs to parse this QS when the component mounts to see if the results match and if not load the correct set.

Currently using `history.pushstate` the QS is updated, but a back button event does not cause the page to render. Instead the result component stays mounted.

If I use router.push with the shallow option, it causes a page refresh which can cause an infinite loop with the ajax calls and the QS updating.
